src/applyMiddleware.js

```diff
@@ -30,7 +30,11 @@ export default function applyMiddleware(...middlewares) {
       getState: store.getState,
       dispatch: (...args) => dispatch(...args)
     }
-    const chain = middlewares.map(middleware => middleware(middlewareAPI))
+    const chain = middlewares.map(middleware => {
+      if (typeof middleware === 'function') {
+        return middleware(middlewareAPI)
+      }
```

Comment:
If middleware is not a function then it'll return `undefined`. The result is a chain contains `undefined`. It throws error when you're trying to compose such a chain.

Reply:
If I use th construction like this:
const middleware = [__DEV__ && createLogger()];

I spent two days to find why my react-native app crashed on start in the production mode...

Reply:
Yes, but this change won't fix your problem. You also didn't add a test, so you wouldn't see this fail.